sdk/core/AzureCore/Source/Pipeline/PipelineStage.swift

```diff
@@ -95,10 +104,10 @@ extension PipelineStage {
                     logger: request.logger,
                     context: request.context
                 )
-                let pipelineError = PipelineError(fromError: error, pipelineResponse: pipelineResponse)
-                self.on(error: pipelineError) { _, handled in
+                let error = AzureError.service("Service error.", error)
```

Comment:
Is this actually always a service error? IIRC, any pipeline stage can return an error. If we update the signature of onRequest / onResponse handlers to return AzureError, we won't need to wrap it in another error here, and the stage itself can determine whether it's an sdk or a service error.

Reply:
Fixed.